libs/s25main/ingameWindows/iwTransport.cpp

```diff
@@ -131,13 +127,11 @@ void iwTransport::Msg_ButtonClick(const unsigned ctrl_id)
 
             GAMECLIENT.visual_settings.transport_order = GAMECLIENT.default_settings.transport_order;
 
-            //// Tooltips in der der Standardbelegung
-            // memcpy(tooltip_indices,STD_TOOLTIP_INDICES,14*sizeof(unsigned short));
-
-            for(unsigned char i = 0; i < 14; ++i)
+            for(unsigned char i = 0; i < buttonData.size(); ++i)
             {
-                group->GetCtrl<ctrlImageButton>(i)->SetImage(TRANSPORT_SPRITES[i]);
-                group->GetCtrl<ctrlImageButton>(i)->SetTooltip(_(TOOLTIPS[i]));
+                const auto& data = buttonData[i];
```

Comment:
its a pity that you can't get the auto-loop-index somehow :-D (on `for(x : list)`)

Reply:
Hm, you can: boost::adapters::indexed IIRC :)

Reply:
so we could do this instead there? :-D

Reply:
I was thinking about it (used it before), but what I really dislike about it (or Boost.Range) is that it is still using C++98. Checked that header and it has includes to Boost.Tuple and Boost.MPL, both are heavy IIRC. And then there is this:

```
// Why yet another "pair" class:
// - std::pair can't store references
```

Which isn't correct. So well, for the sake of compiletimes I'd rather not use it

Reply:
yes that makes sense